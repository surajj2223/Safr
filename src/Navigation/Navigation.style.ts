import styled from 'styled-components/native';

export const Container = styled.SafeAreaView`
  flex: 1;
  background-color: ${({ theme }) => theme.primaryColor};
  justify-content: center;
  align-items: center;
`;

export const ThemedText = styled.Text`
  color: ${({ theme }) => theme.ascentColor};
  font-size: 16px;
`;
