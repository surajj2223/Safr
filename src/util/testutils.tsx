import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import themes, { Theme } from './theme';

interface ThemeWrapperProps {
  children: React.ReactNode;
  theme?: Theme;
}

export const ThemeWrapper: React.FunctionComponent<ThemeWrapperProps> = ({
  children,
  theme = themes.light,
}: ThemeWrapperProps) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
