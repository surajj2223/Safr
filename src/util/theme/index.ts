import light from './light';
import dark from './dark';

export interface Theme {
  primaryColor: string;
  ascentColor: string;
}

export default {
  light,
  dark,
};
