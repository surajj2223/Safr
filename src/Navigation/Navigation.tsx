import React from 'react';
import { Container, ThemedText } from './Navigation.style';

export default () => {
  return (
    <Container>
      <ThemedText>Hi there!</ThemedText>
    </Container>
  );
};
