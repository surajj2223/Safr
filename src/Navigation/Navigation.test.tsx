import React from 'react';
import { render } from '@testing-library/react-native';

import Navigation from './Navigation';
import { ThemeWrapper } from '../util/testutils';

describe('Navigation', () => {
  test('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeWrapper>
        <Navigation />
      </ThemeWrapper>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
