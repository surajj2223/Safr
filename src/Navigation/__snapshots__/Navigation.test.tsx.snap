// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#67AC90",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
/>
`;
